scripts/pylib/twister/twisterlib/reports.py

```diff
@@ -354,6 +354,12 @@ def json_report(self, filename, version="NA", platform=None, filters=None):
             elif instance.status == TwisterStatus.SKIP:
                 suite["status"] = TwisterStatus.SKIP
                 suite["reason"] = instance.reason
+            elif instance.status == TwisterStatus.NOTRUN:
+                suite["status"] = TwisterStatus.NOTRUN
+                suite["reason"] = instance.reason
```

Comment:
this line need to removed to pass the test_report.py blackbox testing, as it expect to be no reason when NOTRUN

Reply:
In my opinion, the errors were caused by an assumption in the `test_fottprint.py`. Proper usage of the NOTRUN status has broken that assumption, so I've fixed the test.